LearningMissionSimulation/PlaygroundEdvard.cs

```diff
@@ -191,12 +193,44 @@ public List<Module> GenerateModuleList()
 
         public void CreateClassrooms(int classroomCount)
         {
-            throw new NotImplementedException();
+            for(int i = 0; i < classroomCount; i++)
+            {
+                Guid moduleId = moduleIdList[AttributeGenerator.random.Next(0, moduleIdList.Count)];
+
+                Module module;
+
+                moduleDictionary.TryGetValue(moduleId, out module);
+
+                Classroom classroom = ObjectGenerator.GenerateClassroom(module);
+
+                classroomDictionary.Add(classroom.Id, classroom);
+            }
+        }
+
+        public void AssignInstructorsToClassrooms()
+        {
+            foreach (Classroom classroom in classroomDictionary.Values)
+            {
+                CandidateFound(classroom);
```

Comment:
1. AssignInstructor is a more appropriate name for your method.
2. You still have three nested loops:
First loop: foreach (Classroom classroom in classroomDictionary.Values)
Second loop: foreach (Instructor instructor in instructorList)
Third loop: if (instructor.ModuleList.Contains(classroom.Module))